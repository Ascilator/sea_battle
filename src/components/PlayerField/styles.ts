import styled from 'styled-components';

export const StyledField = styled.div`
  width: 40vw;
  height: 40vw;
  user-select: none;
`;

export const StyledCell = styled.div`
  flex: 0 1 calc(100% / 12);
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
`;

export const StyledCoords = styled.div`
  display: flex;
  align-items: stretch;
  width: 100%;
  height: calc(100% / 12);
`;

export const StyledFieldCont = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
`;

export const StyledBtnContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 20px;
  margin-top: 20px;
`;
