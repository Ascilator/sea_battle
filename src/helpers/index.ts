export * from './socket';
export * from './components';
export * from './testHelper';
