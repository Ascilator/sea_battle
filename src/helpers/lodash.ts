export const deepClone = (array: Array<any>): Array<any> => {
  const result: Array<any> = [];

  array.forEach(el => {
    if (!Array.isArray(el)) {
      result.push(el);
    } else {
      result.push(deepClone(el));
    }
  });

  return result;
};
