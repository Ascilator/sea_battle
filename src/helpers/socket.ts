import io from "socket.io-client";

import { BASE_URL_FOR_SOCKETS } from "@/constants";

const RECONNECT_ATTEMPTS: number = 2;

const socket = io(BASE_URL_FOR_SOCKETS, {
  path: `${""}/socket.io/`,
  transports: ["websocket"],
  reconnection: true,
  reconnectionAttempts: RECONNECT_ATTEMPTS,
  reconnectionDelay: 1000,
  reconnectionDelayMax: 5000,
  autoConnect: true,
});

export { socket };
