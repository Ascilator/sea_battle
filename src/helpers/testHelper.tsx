import { render as rtlRender } from '@testing-library/react';
import { FC, ReactElement } from 'react';
import { Provider } from 'react-redux';
import { Store } from 'redux';

export const renderComponentForTest = (
  component: ReactElement,
  {
    store,
    ...renderOptions
  }: {
    store: Store;
  }
) => {
  //  <Provider store={store}>

  const Wrapper: FC = ({ children }) => <Provider store={store}>{children}</Provider>;

  return rtlRender(component, { wrapper: Wrapper, ...renderOptions });
};

export * from '@testing-library/react';
