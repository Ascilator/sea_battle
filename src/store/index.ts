import { configureStore } from '@reduxjs/toolkit';

import canClickReducer from './canClick';

export const store = configureStore({
  reducer: {
    canClick: canClickReducer
  }
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
