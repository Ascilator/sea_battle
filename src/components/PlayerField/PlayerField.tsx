import { FC, useState } from 'react'
import { FieldRow } from '@/components/FieldRow'
import { alphabet, matrix } from '@/constants'
import { FieldState } from '@/types'

import { StyledField, StyledCoords, StyledCell } from './styles'

const PlayerField: FC = () => {
  const [gameState] = useState<FieldState>(matrix)

  const renderRows = () => {
    return gameState.map((row, index) => <FieldRow key={index} value={index + 1} rowData={row} />)
  }

  const renderCoords = () => {
    return (
      <StyledCoords>
        <StyledCell />
        {alphabet.map((el) => (
          <StyledCell key={el}>{el}</StyledCell>
        ))}
        <StyledCell />
      </StyledCoords>
    )
  }
  return (
    <StyledField>
      {renderCoords()}
      {renderRows()}
      {renderCoords()}
    </StyledField>
  )
}

export { PlayerField }
