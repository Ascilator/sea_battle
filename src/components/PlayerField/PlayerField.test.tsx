import { renderComponentForTest } from '@/helpers';
import { store } from '@/store';

import { PlayerField } from './PlayerField';

test('PlayerField component', () => {
  const playerField = renderComponentForTest(<PlayerField />, {
    store
  });

  expect(playerField).toMatchSnapshot();
});
