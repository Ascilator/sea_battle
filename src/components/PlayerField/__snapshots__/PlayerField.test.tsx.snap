// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerField component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iCfMLu kwmnDz"
      >
        <div
          class="sc-eCImPb fEIBIZ"
        >
          <div
            class="sc-gKclnd giphph"
          >
            <div
              class="sc-jRQBWg jJWUFM"
            />
            <div
              class="sc-jRQBWg jJWUFM"
            >
              a
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              b
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              c
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              d
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              f
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              g
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              h
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              i
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              j
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              k
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            />
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              1
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              1
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              2
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              2
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              3
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              3
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              4
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              4
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              5
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              5
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              6
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              6
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              7
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              7
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              8
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              8
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              9
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              9
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              10
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              10
            </div>
          </div>
          <div
            class="sc-gKclnd giphph"
          >
            <div
              class="sc-jRQBWg jJWUFM"
            />
            <div
              class="sc-jRQBWg jJWUFM"
            >
              a
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              b
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              c
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              d
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              f
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              g
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              h
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              i
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              j
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            >
              k
            </div>
            <div
              class="sc-jRQBWg jJWUFM"
            />
          </div>
        </div>
        <div
          class="sc-furwcr fEGCYS"
        >
          <button
            class="sc-hKwDye blNMfZ"
            color="#FFCC00"
          >
            <span>
              Change position
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iCfMLu kwmnDz"
    >
      <div
        class="sc-eCImPb fEIBIZ"
      >
        <div
          class="sc-gKclnd giphph"
        >
          <div
            class="sc-jRQBWg jJWUFM"
          />
          <div
            class="sc-jRQBWg jJWUFM"
          >
            a
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            b
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            c
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            d
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            f
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            g
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            h
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            i
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            j
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            k
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          />
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            1
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            1
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            2
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            2
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            3
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            3
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            4
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            4
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            5
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            5
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            6
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            6
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            7
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            7
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            8
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            8
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            9
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            9
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            10
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            10
          </div>
        </div>
        <div
          class="sc-gKclnd giphph"
        >
          <div
            class="sc-jRQBWg jJWUFM"
          />
          <div
            class="sc-jRQBWg jJWUFM"
          >
            a
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            b
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            c
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            d
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            f
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            g
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            h
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            i
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            j
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          >
            k
          </div>
          <div
            class="sc-jRQBWg jJWUFM"
          />
        </div>
      </div>
      <div
        class="sc-furwcr fEGCYS"
      >
        <button
          class="sc-hKwDye blNMfZ"
          color="#FFCC00"
        >
          <span>
            Change position
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
